### PR TITLE
validations and model specs

### PR DESCRIPTION
**Pull request by erickma1**
In this milestone, I did the following;

- Set validation rule for users and Posts
- Test case for User Model
- Test case for Post Model
- Test case for Comment Model
- Test case for Like Model

### CODE REVIEW

File: spec/models/comment_spec.rb
Comment (Efiamotu-1) [inline spec/models/comment_spec.rb:1]: - [ ] As shown in the image below your test is not passing, please kindly fix it accordingly :+1: 
fc

File: spec/models/like_spec.rb
Comment (Efiamotu-1) [inline spec/models/like_spec.rb:1]: - [ ] As shown in the image below your test is not passing, please kindly fix it accordingly :+1: 
lc

File: spec/models/post_spec.rb
Comment (Efiamotu-1) [inline spec/models/post_spec.rb:1]: - [ ] As shown in the image below your test is not passing, please kindly fix it accordingly :+1: 
pc

File: spec/models/user_spec.rb
Comment (Efiamotu-1) [inline spec/models/user_spec.rb:1]: - [ ] Please kindly add the below test to your user test case :+1: 
co

File: spec/models/post_spec.rb
Comment (Efiamotu-1) [inline spec/models/post_spec.rb:1]: - [ ] kindlyd add the below test for your post model :+1: 
 
vd

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @erickma1, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by erickma1** [comment]
- Corrected errors related to tests
- Added required validations

File: spec/models/post_spec.rb
Comment (VICTOR-CHUKWU) [reply spec/models/post_spec.rb:1]: - [ ] Just as the previous reviewer mentioned kindly include a test that will test if likes_counter is an integer, you have already done similar test for comments_counter but none for likes_counter, kindly fix this as it is the only test you seem to be missing at the moment :+1:

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @erickma1 , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by erickma1** [comment]
include a test that will test if likes_counter is an integer

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -25,6 +25,12 @@
     expect(post).to_not be_valid
   end
 
+  it 'is not valid with a negative likes_counter' do
+    user = User.create(name: 'John', posts_counter: 0)
+    post = user.posts.build(title: 'Negative Likes', comments_counter: -1, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
```
Comment (Gambit142) [inline spec/models/post_spec.rb]: - [ ] Great work attempting to test the validity of the `likes_counter` attribute of the `Post` model. However, in the newly written test, you have only tested the `comments_counter` and not the `likes_counter`. Therefore, you have written two of a kind for the `comments_counter`. 👍 💻 

#### N.B: Please ensure that all tests pass after making the necessary corrections. 🖥️ 🖱️

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @erickma1 👋, 

Great work on making the changes requested by a previous reviewer 👏 

## Highlights
- Setup linters correctly ✔️ 
- Validated different model attributes ✔️ 
- Wrote model spec tests for different models ✔️ 

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by erickma1** [comment]
include a test that will test if likes_counter is an integer

**Review by divinecharlotte — APPROVED** [review]
# Hello @erickma1      😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 
### To Highlight 💯

- Good implementation of the required changes✔️
- Descriptive pr✔️






## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
